Source/DiabloUI/selgame.cpp

```diff
@@ -412,10 +414,11 @@ static bool IsGameCompatible(GameData *data)
 	if (data->programid != GAME_ID) {
 		UiSelOkDialog(title, "The host is running a different game than you.", false);
 	} else {
-		char msg[64];
-		sprintf(msg, "Your version %s does not match the host %d.%d.%d.", PROJECT_VERSION, PROJECT_VERSION_MAJOR, PROJECT_VERSION_MINOR, PROJECT_VERSION_PATCH);
-
-		UiSelOkDialog(title, msg, false);
+		// An example of using a format with references. Note the references that allow a localized string to change the order of the inserted parameters.
+		// Note that named insertions would probably be better here, but this is just to provide an example.
+		auto msg = fmt::format("Your version {0} does not match the host {1}.{2}.{3}.", PROJECT_VERSION, PROJECT_VERSION_MAJOR, PROJECT_VERSION_MINOR, PROJECT_VERSION_PATCH);

```

Comment:
This is _so_ much better. No more fixed array shenanigans.

Reply:
How would this work if you wanted to pad a number? E.g.
`printf("The percent is %02d", x);`

Reply:
@galaxyhaxz, `fmt::print("The percent is {:02d}", x);`
See: https://fmt.dev/latest/syntax.html#format-examples